backend/inspirehep/disambiguation/tasks.py

```diff
@@ -105,16 +105,30 @@ def assign_reference_to_author_if_unambiguous_match(matched_authors_refs, author
 def assign_reference_to_author_if_unambiguous_literature_author_match(
     matched_records, author
 ):
+    matched_record_list = list(matched_records)
     matched_authors_refs = get_record_refs_for_matched_literature_authors(
-        matched_records
+        matched_record_list
     )
     matched_ref = assign_reference_to_author_if_unambiguous_match(
         matched_authors_refs, author
     )
     if matched_ref:
+        author_ids = get_value(
+            matched_record_list[0], "inner_hits.authors.hits.hits._source.ids", []
```

Comment:
what if `len(matched_record_list) == 0`?

Reply:
then there will be no matched_ref

Reply:
It will raise an `IndexError` though

Reply:
No, because `matched_ref` is True only in case when there is one author match. 
https://github.com/inspirehep/inspirehep/blob/1aefc405c244823a29ca7b40f8555831abf52097/backend/inspirehep/disambiguation/tasks.py#L98-L102
So if there is no match reference can't be assigned (and `None` is returned from `assign_reference_to_author_if_unambiguous_match`) and this part of the code won't be executed.